src/main/java/domen/rideapp/infrastructure/AppConfig.java

```diff
@@ -38,4 +57,47 @@ RideService rideService(PricingService pricingService,
     DriverService driverService(DriverRepository repository) {
         return new DriverService(repository);
     }
+
+    @Bean
+    @ConfigurationProperties(prefix = "google.maps")
+    public GoogleMapsConfig googleMapsConfig() {
+        return new GoogleMapsConfig();
+    }
+
+    @Profile("prod")
+    @Bean
+    MapService googleMapsService(GoogleMapsClient client,
+                                 ConversionService conversionService,
+                                 GoogleMapsConfig config) {
+        if (config.getApiKey() == null || config.getApiKey().isBlank()) {
+            throw new IllegalStateException("Google Maps API key must be set in production.");
+        }
+        return new GoogleMapsService(client, conversionService);
+    }
+
+    @Profile({"test", "dev"})
```

Comment:
Kod źródłowy nie powinien mieć profilu "test", co najwyżej "local". Jeśli testy wymagają tego profilu to trzeba stworzyć odpowiedni bean w katalogu testów